### PR TITLE
Adding slices

### PR DESCRIPTION
**Pull request by djdiskmachine**
Rebased on master
Issue when loading project "restoring allocation" whenever 164-170 are added

### CODE REVIEW

**Review by democloid — CHANGES_REQUESTED** [review]
Regarding the crash, have you tried to create a new project to test this?

File: sources/Application/Instruments/SampleInstrument.h [hunk 1]
```diff
@@ -36,6 +37,7 @@ enum SampleInstrumentLoopMode {
 #define SIP_LOOPLEN MAKE_FOURCC('L', 'L', 'E', 'N')
 #define SIP_INTERPOLATION MAKE_FOURCC('I', 'N', 'T', 'P')
 #define SIP_SAMPLE MAKE_FOURCC('S', 'M', 'P', 'L')
+#define SIP_SLICES MAKE_FOURCC('S','L','C','S')
```
[the remark this answers is not in the document]
Comment (djdiskmachine) [reply sources/Application/Instruments/SampleInstrument.h:40]: Yes working on it!

File: sources/Application/Views/InstrumentView.cpp [hunk 2]
```diff
@@ -161,6 +161,14 @@ void InstrumentView::fillSampleParameters() {
...
   fieldList_.insert(fieldList_.end(), &(*intVarField_.rbegin()));
 
+  if (v->GetString() == "slicer") {
+    position._x += 18;
+    v = instrument->FindVariable(SIP_SLICES);
+    intVarField_.emplace_back(position, *v, "slcs: %d", 1, 128, 1, 16);
+    fieldList_.insert(fieldList_.end(), &(*intVarField_.rbegin()));
+    position._x -= 18;
+  }
+
```
[the remark this answers is not in the document]
Comment (djdiskmachine) [reply sources/Application/Views/InstrumentView.cpp:171]: Yes, It's added conditionally but it's not perfect
You have to exit then enter the instrument screen for it to refresh, I can't figure out how to make it refresh when selecting slicer
PXL_20240403_120826428 MP~2
PXL_20240403_120223122 MP~3

File: docs/MANUAL.md [hunk 3]
```diff
@@ -302,6 +302,7 @@ TODO: screencap
   - loop will start at zero and loop from loopstart to end.
   - looper sync will automatically tune a loop so that it plays exactly 16 bars. Use the root note to play twice faster/slower
   - oscillator is a special mode where the loop selection (from loopstart to end) is taken as oscillator data and automatically tuned. Experiment with different settings, do not forget 'root note' is your friend to tune the oscillator back in a useful range
+  - slicer will cut the sample into "slices" (displayed only in slice mode) amount of samples. Slices are mapped from C-2 (the lowest note) up to amount of slices. Example: slices == 4 will give you four slices mapped to C-2, C#-2, D-2, D#-2
```
Comment (democloid) [inline docs/MANUAL.md:305]: @djdiskmachine Given that we have a "Root note" field, what do you think using that as the root of the slices? Or is it understood than any use of slices generally would start at C-2?

**Comment by djdiskmachine** [comment]
@democloid rootnote will still affect the playback pitch. I made the implementation to be the same as on M8tracker, setting the first slice to C3 will make 48 the max number of slices. :)

**Comment by djdiskmachine** [comment]
@democloid is this still relevant or should i delete it?

**Comment by maks** [comment]
@djdiskmachine please dont delete it! I'd really love to have slicing support in pT!
Sorry about this sitting untouched for so long! It will likely need a bit of rework onto the changes that are being done in the pT codebase, but I'm happy to put my hand up to help out with that.

**Pull request closed without merging**